bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -1014,26 +1034,28 @@ private void executeActions(Rule rule, boolean stopOnFirstFail) {
                 }
             }
         }
-
     }
 
     /**
      * The method clean used resource by rule engine when it is stopped.
      */
-    public synchronized void dispose() {
-        if (!isDisposed) {
-            isDisposed = true;
-            for (Iterator<RuntimeRule> it = rules.values().iterator(); it.hasNext();) {
-                RuntimeRule r = it.next();
-                removeRuleEntry(r);
-                it.remove();
-            }
-            if (compositeFactory != null) {
-                compositeFactory.dispose();
-                compositeFactory = null;
+    @Deactivate
+    public void dispose() {
```

Comment:
If this is now only called from DS, I would rename it to `deactivate` and make it protected.

Reply:
OK